kb_sdk_plus.md

```diff
@@ -1,10 +1,98 @@
 # KB-SDK plus
 
+## Why?
+
+The KBase SDK is a core piece of infrastructure, but it has become exceedingly difficult to
+maintain and expand as the test suite is exceptionally complex and currently does not pass.
+As such, large changes to the codebase are risky and no updates have been released for 5+ years.
+Furthermore, due to outdated dependencies, the SDK will not run on modern machines.
+
+To rectify this while still supporting older apps, we have elected to fork the
+https://github.com/kbase/kb_sdk repo to `kb_sdk_plus`. Version 1.2.1 is the last `kb_sdk`
+release and will be kept as is. Future SDK development will occur in `kb_sdk_plus`.
+
+## Why not kb_sdk2?
+
+`<repo>2` generally signifies a greenfield rewrite of a codebase in KBase,
+which `kb_sdk_plus` is not. It's forking a codebase to leave the original codebase untouched
+and runnable while we make changes to the forked codebase.
+
+## Guidelines
+
+* Backwards incompatible changes are expected in some respects:
+  * R code generation will be removed
+    * Unused in KBase or anywhere else as far as we know
+  * Perl code generation will almost certainly be removed
+    * Perl apps can continue to be supported with the original SDK, or wrapped with
+      Python / Java
+      * https://github.com/boriel/perlfunc for instance
+* Otherwise `kb_sdk_plus` needs to be backwards compatible with `kb_sdk`
+  * It must work with the current KBase backend services
+  * Python and Java apps must continue to be supported
+    * It might be possible to make backwards incompatible changes that are fixed
+      with a recompile of an app or a simple migration guide.
+      This would need careful thought and implementation.
+* Generally simplify the codebase where possible and document where needed
+      
+## Prior to releasing for general development
+
+* Get the tests to pass locally
+  * Remove R support, probably remove Perl support
+  * Do we still need Javascript support?
```

Comment:
@briehl @ialarmedalien wdyk?

Reply:
What does "support" mean here? Just making clients? We have a bunch of different implementations of JS clients throughout the codebase, made for different purposes.

I'd like to normalize on one, but that feels like feature creep here, especially as the current SDK JS client code consists of an AJAX call + a name string and very little type checking or error handling.

I think I just convinced myself that we don't need to support it in this particular effort.

Reply:
Also, once things are spruced up, it probably won't be a heavy lift to tack on JS client support, if that's desired.

Reply:
My understanding is that the JS clients at this point are effectively generic JSONRPC clients, so there's no real reason for the SDK to generate anything